Src/FluentAssertions/Equivalency/Steps/EnumerableEquivalencyStep.cs

```diff
@@ -61,10 +61,15 @@ internal static object[] ToArray(object value)
         {
             return ((IEnumerable)value).Cast<object>().ToArray();
         }
-        catch (InvalidOperationException) when (value.GetType().Name.Equals("ImmutableArray`1", StringComparison.Ordinal))
+        catch (InvalidOperationException) when (IsIgnorableArrayLikeType(value))
         {
-            // This is probably a default ImmutableArray<T>
+            // This is probably a default ImmutableArray<T> or an empty ArraySegment.
             return Array.Empty<object>();
         }
+
+        bool IsIgnorableArrayLikeType(object value)
```

Comment:
🔧 We prefer to use normal methods instead of local functions
🙃 It might be me, but I prefer normal methods over expression-bodied methods